microsoft/testsuites/acctests/sgxtest.py

```diff
@@ -0,0 +1,217 @@
+    area="ACC",
+    category="functional",
+    owner="acc@microsoft.com",
+    description="""
+    This is a part of ACC test suite.
+    This test case validates the installation of SGX. 
+    The Suite consists of 2 sets of tests:
+    1. High priority Level 0 test: verifyOE
+    2. Lower Priority Level 1 test: runAllOESamples 
+
+    verifyOE: Runs basic tests that validate SGX platform. 
+    runAllOESamples: Runs all samples
+    """,
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class SgxTestSuite(TestSuite):
+    #This is a run_acc_tests.sh script initializer. 
+    #run_acc_tests.sh installs and sets up OE 
+    # On the node
+    #To Do: Remove unnecessary tests
+    def before_suite(self, log: Logger, **kwargs: Any) -> None:
+        self._acctest_script = CustomScriptBuilder(
+            Path(__file__).parent.joinpath("scripts"), ["run_acc_tests.sh"]
+        )
+    TIMEOUT = 2000
+    @TestCaseMetadata(
+        description="""
+        This test case validates functionality of Secure Guard Extensions platform.
+            1. Validate if OpenEnclave is installed on the Node
+            2. Get SGX version
+            3. Perform Remote Attestation and Hello world on the Node
+        """,
+        priority=0,
+        timeout=TIMEOUT,
+        use_new_environment=False,
+    )
+    def verifyOE(self, node: Node, log: Logger) -> None:
+        #GET GUEST OS VERSION
+        if node.os.is_posix:
+            assert isinstance(node.os, Posix)
+            info = node.tools[Uname].get_linux_information()
+            log.info(
+                f"release: '{info.uname_version}', "
+                f"version: '{info.kernel_version_raw}', "
+                f"hardware: '{info.hardware_platform}', "
+                f"os: '{info.operating_system}'"
+            )
+        else:
+            log.info("windows operating system")
+
+        #CHECK IF OE IS INSTALLED
+        script: CustomScript = node.tools[self._acctest_script]
+        result1 = script.run()
+        python = node.tools[Python]
+
+        openenclave_install_path = python.run(
+            f'-c "os.path.expanduser("~/openenclave-install/")"',
+            force_run=True,
+        )       
+        
+        if openenclave_install_path:
+            #Run oesgx tool that comes with OESDK to get the version
+            oesgx = node.execute([openenclave_install_path + "bin/oesgx"], shell=True)
+            sgx_version = ""
+            if "CPU supports Software Guard Extensions:SGX1" in oesgx:
+                sgx_version = "sgx1"
+            elif "CPU supports Software Guard Extensions:SGX2" in oesgx:
+                sgx_version = "sgx2"
+            log.info(sgx_version)
+            assert sgx_version
+
+            #RUN REMOTE ATTESTATION
+            SAMPLES = [
+            "helloworld",
+            "attestation"
+            ]
+            returncode = runOESamples(SAMPLES)
+            assert returncode == 0
+
+            #CHECK PROCESS OUTPUT.
```

Comment:
It doesn't need like the sample code. You can use result.assert_exit_code to assert the result.